cmd/thanos/receive.go

```diff
@@ -294,15 +290,68 @@ func runReceive(
 		})
 	}
 
-	level.Debug(logger).Log("msg", "setting up grpc server")
+	level.Debug(logger).Log("msg", "setting up gRPC server")
 	{
-		if err := setupAndRunGRPCServer(g, logger, reg, tracer, conf, reloadGRPCServer, comp, dbs, webHandler, grpcLogOpts,
```

Comment:
Why not keeping this handy function to set up the grpc server? I think it was helping a lot with the organization of the code and overall readability of the `runReceive` function.

Reply:
I kept it similar to how we set up HTTP server, i.e. we do it in a separate code block, but I removed this function since it feels like we're just passing lot of parameters while the function is used only once and it's still somewhat isolated from the rest of the code by the code block, so that helps readability. I'd be open to more / larger improvements as part of a bigger refactoring in https://github.com/thanos-io/thanos/issues/5643.

Reply:
Got it. I miss it because if we didn't have that debug log just above the block, there would be nothing to help me as a reader quickly have an idea of what this code block does. 

It makes my life so much easier when I am scrolling through `runReceive` trying to understand everything it does if there are a few functions with good names being called instead of a bunch of code blocks without a clear description of intent besides a `log` call right before the block.

Reply:
That's totally a legitimate concern :+1: Let's tackle this in that refactor.